pkg/config/autodiscover/csi_info.go

```diff
@@ -0,0 +1,58 @@
+package autodiscover
+
+import (
+	"fmt"
+	"strings"
+)
+
+// PodSetList holds the data from an `oc get deployment/statefulset -o json` command
+
+// PodSetResource defines deployment/statefulset resources
+var (
+	csiCommand          = "oc get csidriver -o go-template='{{ range .items}}{{.metadata.name}} {{end}}'"
+	depNameCommand      = "oc get pods -A -o go-template='{{ range .items}}{{ $alllabels := .metadata.labels}}{{ $namespace := .metadata.namespace}}{{ range .spec.containers }}{{ range .args }}{{if eq . \"--driver-name=%s\"}}{{ range $label,$value := $alllabels}}{{if eq $label \"app.kubernetes.io/managed-by\"}}{{$value}} {{$namespace}}{{end}}{{end}}{{end}}{{end}}{{end}}{{end}}'"
```

Comment:
The workflow to get the operator name/organization does now work for every operator in every situation. For instance, if the controller pods are not deployed, we cannot get the deployment name. Instead of current script I propose that we use a mapping between operator provided API and CSI driver name. The CSI driver name and the CRI kind are well known (see https://raw.githubusercontent.com/kubernetes-csi/docs/master/book/src/drivers.md) so they will provide a stable mapping. Will work on code to change this

Reply:
Daft proposal is at :
https://github.com/aabughosh/test-network-function/pull/1